### PR TITLE
Adds client ResourceProvider for ServletContext

### PR DESCRIPTION
**Pull request by lordofthejars**
Resolves issue https://issues.jboss.org/browse/ARQ-1973

### CODE REVIEW

File: jetty-embedded-6.1/src/main/java/org/jboss/arquillian/container/jetty/embedded_6_1/JettyEmbeddedContainer.java [hunk 1]
```diff
@@ -106,6 +109,10 @@
     @DeploymentScoped
     private InstanceProducer<WebAppContext> webAppContextProducer;
 
+    @Inject
+    @ApplicationScoped
+    private InstanceProducer<ServletContext> servletContextInstanceProducer;
```
Comment (aslakknutsen) [inline jetty-embedded-6.1/src/main/java/org/jboss/arquillian/container/jetty/embedded_6_1/JettyEmbeddedContainer.java]: This should be DeploymentScoped

Comment (lordofthejars) [inline jetty-embedded-6.1/src/main/java/org/jboss/arquillian/container/jetty/embedded_6_1/JettyEmbeddedContainer.java]: Changed, I was not sure what was the difference between both scopes in terms of extensions.

File: jetty-common/src/main/java/org/jboss/arquillian/container/jetty/ServletContextResourceProvider.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import javax.servlet.ServletContext;
+import java.lang.annotation.Annotation;
+
+/**
+ * ServletContextResourceProvider
+ *
+ * @author asotobu
+ */
+public class ServletContextResourceProvider implements ResourceProvider {
+
```
Comment (aslakknutsen) [inline jetty-common/src/main/java/org/jboss/arquillian/container/jetty/ServletContextResourceProvider.java]: This class should be part of the change in arquillian-core, as a 'Provide anything produced on Client side'

File: jetty-common/src/main/java/org/jboss/arquillian/container/jetty/ServletContextResourceProvider.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    @Override
+    public boolean canProvide(Class<?> type) {
+        return javax.servlet.ServletContext.class.isAssignableFrom(type);
+    }
+
+    @Override
+    public Object lookup(ArquillianResource arquillianResource, Annotation... annotations) {
+
+        if(servletContextInstance.get() == null) {
```
Comment (aslakknutsen) [inline jetty-common/src/main/java/org/jboss/arquillian/container/jetty/ServletContextResourceProvider.java]: returning null here is fine. Core will handle exception on null.

File: jetty-embedded-6.1/src/main/java/org/jboss/arquillian/container/jetty/embedded_6_1/JettyExtension.java [hunk 4]
```diff
@@ -31,6 +33,7 @@
    public void register(ExtensionBuilder builder)
    {
       builder.service(DeployableContainer.class, JettyEmbeddedContainer.class);
+      builder.service(ResourceProvider.class, ServletContextResourceProvider.class);
```
Comment (aslakknutsen) [inline jetty-embedded-6.1/src/main/java/org/jboss/arquillian/container/jetty/embedded_6_1/JettyExtension.java]: Should be part of arquillian-core/servlet-protocol

File: jetty-common/src/main/java/org/jboss/arquillian/container/jetty/ServletContextResourceProvider.java (hunk 2 above)
Comment (aslakknutsen) [inline jetty-common/src/main/java/org/jboss/arquillian/container/jetty/ServletContextResourceProvider.java]: You probably want to make this extend the OperatesOnDeploymentAwareProvider to allow ServletContexts from multiple Deployment be injected when operating in the other context. @ArquillianResource @OperatesOnDeployment("X")

https://github.com/arquillian/arquillian-core/blob/master/container/test-impl-base/src/main/java/org/jboss/arquillian/container/test/impl/enricher/resource/OperatesOnDeploymentAwareProvider.java

**Comment by aslakknutsen** [comment]
Thank you! Pushed upstream https://github.com/arquillian/arquillian-container-jetty/commit/ce138139af3083ec42417d40912ffb467bdf1980

**Pull request closed without merging**